examples/synchronization/multi_function/cont_ai_ci_tdms_sync.py

```diff
@@ -0,0 +1,240 @@
+"""Example of logging multiple synchronized tasks to a single TDMS file using a queue.
+
+This example demonstrates how to:
+1. Synchronize multiple DAQmx tasks using a shared clock
+2. Use a producer-consumer pattern with a queue
+3. Log data from multiple tasks to a single TDMS file
+"""
+
+import os
+import queue
+import threading
+import time
+from collections.abc import Sequence
+from concurrent.futures import ThreadPoolExecutor
+from pathlib import Path
+from typing import List, Union
+
+from nitypes.waveform import AnalogWaveform
+from nptdms import ChannelObject, GroupObject, RootObject, TdmsFile, TdmsWriter
+
+import nidaqmx
+from nidaqmx.constants import (
+    AcquisitionType,
+    Edge,
+)
+from nidaqmx.stream_readers import AnalogMultiChannelReader, CounterReader
+
+os.environ["NIDAQMX_ENABLE_WAVEFORM_SUPPORT"] = "1"
+
+# Configuration
+SAMPLE_RATE = 1000
+SAMPLES_PER_CHANNEL = 1000
+TIMEOUT = 10.0
+# Note: This example currently assumes all tasks run on a single synchronized device.
+# To make this example work across multiple devices, export
+# a shared sample clock from one device to the other via a PFI line.
+DEVICE_NAME = "Dev1"
+
+TaskData = Union[
+    List[AnalogWaveform],  # Analog input: list of waveforms
+    AnalogWaveform,  # Counter input: waveform
+    None,  # Sentinel value
+]
+
+data_queue: queue.Queue[List[TaskData]] = queue.Queue(maxsize=10)
+
+
+def producer(
+    tasks: Sequence[nidaqmx.Task],
+    data_queue: queue.Queue[List[TaskData]],
+    stop_event: threading.Event,
+) -> None:
+    """Producer function that reads data from DAQmx tasks and puts it in the queue."""
+    # The queue holds a list with task data and timing:
+    # [
+    #   [AnalogWaveform, AnalogWaveform],  # Element 0: AI data - list of waveform objects
+    #   AnalogWaveform,                    # Element 1: Counter data - single waveform object
+    # ]
+    ai_reader = AnalogMultiChannelReader(tasks[0].in_stream)
+    counter_reader = CounterReader(tasks[1].in_stream)
+
+    num_ai_channels = len(tasks[0].ai_channels.all)
+
+    try:
+        while not stop_event.is_set():
+            data = []
+
+            ai_waveforms = [
+                AnalogWaveform(sample_count=SAMPLES_PER_CHANNEL) for _ in range(num_ai_channels)
+            ]
+            counter_waveform = AnalogWaveform(sample_count=SAMPLES_PER_CHANNEL)
+
+            ai_reader.read_waveforms(
+                ai_waveforms, number_of_samples_per_channel=SAMPLES_PER_CHANNEL, timeout=TIMEOUT
+            )
+            data.append(ai_waveforms)
+
+            counter_reader.read_many_sample_double(
+                counter_waveform._data,
+                number_of_samples_per_channel=SAMPLES_PER_CHANNEL,
+                timeout=TIMEOUT,
+            )
+            data.append(counter_waveform)
```

Comment:
Consider filling in the counter waveform's timing and extended properties.

```suggestion
            counter_waveform.timing = ai_waveforms[0].timing
            counter_waveform.channel_name = tasks[1].channel_names[0]
            data.append(counter_waveform)
```